apps/browser/settings/declarativeloginmodel.cpp

```diff
@@ -119,6 +124,8 @@ void DeclarativeLoginModel::handleRecvObserve(const QString &message, const QVar
 {
     if (message == ALL_LOGINS) {
         setLoginList(qvariant_cast<QVariantList>(data));
+        // We're only interested in receiving the message once
+        SailfishOS::WebEngine::instance()->removeObserver(ALL_LOGINS);
```

Comment:
Asking maybe obvious things, but...
- What happens to new login after first set returned? I.e. will to still be visible?
- Is model created each time again when entering to the login settings page?

Reply:
Not obvious, and worth considering.

What you say is correct: the model is recreated each time the user enters the page. We could definitely keep it persistent and perform updates on change notifications from embedlite-components. But I wasn't convinced it was worth keeping it in memory the whole time, plus it avoided the worry of the two (UI model and gecko's model) getting out of sync. I also figured there wouldn't be any situation in which the model could be updated while the page was open (since any change would usually be triggered by some user interaction).

In this respect it differs from the history and bookmark models, but those two have to be kept in memory the whole time as they're used by the toolbar.

If you think it's better to keep it updated, I'm happy to make the change.